### DISCUSSION TITLE
Applying the date range selector to the /budgets pages

### DISCUSSION THREAD

**Discussion by FFSchier** [firefly-iii/firefly-iii#9344] [General]
Hi,
I discovered Firefly III a while ago and only got around to set things up - took a while to get my bank statments to acutally integrate nicely, but now my preprocessing catches just about every funky thing my bank does. Thank you for creating and maintaining Firefly III!

I noticed that the Budget pages do not adhere to the defined date ranges. (This is on Version 6.1.21.)
We can force them through reports to only give information on given date ranges - yet I wondered whether the non-adherence was by design, and if so, why you chose to do so. 
Cheers!

**Comment by JC5** [comment]
Hey, good question! 

It turns out it's a lot easier to use the month selectors to go back and forth on that particular page than it is the date selector at the top. I keep them separate for that reason.

**Reply by FFSchier** [reply]
Ah, yes, I apologize - I described it poorly. 
For the `/budgets` page I entirely agree. That is a lot more intuitive.

I meant the `/budgets/show/xx` pages. Here is a screenshot of the behaviour:

![image](https://github.com/user-attachments/assets/704ccec3-659c-40d6-a5d2-9fbacb2158ff)


Opposite to most (all?) similar-styled pages, this sub-section seems not to adhere to the selected date range, which makes reviewing on a per-budget-level a little tricky at times.
Is this behaviour by design?

**Reply by JC5** [reply]
Yes, if you set a budget amount for a specific time period, you can use those to navigate more fine grained.
